### PR TITLE
Benji/extension components approve account backup

### PR DESCRIPTION
**Pull request by thebkr7**
Closes #ISSUE

**Description:**

3 components for the Chrom extension: Approve Transaction, Account, Backup Codes.

I am working with dummy data here and will need to see how we plan to pass each component it's data.

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [x] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/components/common/SessionApprove.vue [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            ],
+            memo: ""
+          }
+        },
+        timestamp: "2019-06-19T17:00:00Z",
+        type: "staking",
+        time: "2019-06-19T17:00:00.277Z"
+      }
+    ],
+    deligates: [
```
Comment (jbibla) [inline src/components/common/SessionApprove.vue]:
```suggestion
    delegates: [
```

**Comment by thebkr7** [comment]
I will add tests Friday.

File: src/components/transactions/LiTransaction.vue [hunk 2]
```diff
@@ -63,6 +63,10 @@ export default {
     fees: {
       type: Object,
       required: true
+    },
+    hideData: {
```
Comment (faboweb) [inline src/components/transactions/LiTransaction.vue]: Maybe something more descriptive like: `preSend`?

Comment (faboweb) [reply src/components/transactions/LiTransaction.vue]: Or `hideBlockData`

File: src/components/common/SessionApprove.vue [hunk 3]
```diff
@@ -0,0 +1,209 @@
+<template>
+  <TmPage class="approve-tran" hide-header>
```
Comment (faboweb) [inline src/components/common/SessionApprove.vue]: this doesn't need to be a page, this only adds the footer, which we don't need I think

File: src/components/common/SessionAccounts.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
+<template>
+  <TmPage class="accounts" :hide-header="true">
```
Comment (jbibla) [inline src/components/common/SessionAccounts.vue]: no page required

File: src/components/common/SessionBackupCodes.vue [hunk 5]
```diff
@@ -0,0 +1,89 @@
+<template>
+  <TmPage class="backup-code" hide-header>
```
Comment (jbibla) [inline src/components/common/SessionBackupCodes.vue]: no page

File: src/components/transactions/LiTransaction.vue (hunk 2 above)
Comment (jbibla) [reply src/components/transactions/LiTransaction.vue]: maybe `hideRightSide`

Comment (colw) [reply src/components/transactions/LiTransaction.vue]: What's the purpose on this scenario to hide the fees? Can't we jsut check the fees prop?

Comment (jbibla) [reply src/components/transactions/LiTransaction.vue]: the fees are visible below. we're also hiding the block number and datetime.

Screen Shot 2019-06-21 at 8 44 18 AM

**Comment by thebkr7** [comment]
I also need to update the address in the transaction to be shortened, and under the transaction to show the from account.

**Comment by colw** [comment]
Should we close this in favour of the `chrome-extension` branch?

**Comment by thebkr7** [comment]
Working on tests. Please don't merge yet. Or do and I can just switch branches

**Comment by thebkr7** [comment]
Tests for the three components updated. But pre-existing tests from Chrome-Extension branch are still failing.

File: src/components/transactions/LiAnyTransaction.vue [hunk 6]
```diff
@@ -10,7 +10,7 @@
         :time="transaction.time"
         :block="Number(transaction.height)"
         :memo="transaction.tx.value.memo"
-        :hide-data="hideData"
+        :hide-right-side="hideRightSide"
```
Comment (colw) [inline src/components/transactions/LiAnyTransaction.vue]: What is the right side, and why are we hiding it?

Comment (thebkr7) [reply src/components/transactions/LiAnyTransaction.vue]: It's the right side of the transaction components that displays block #, Time/date etc. It adds unnecessary noise to the extension IMO

Comment (colw) [reply src/components/transactions/LiAnyTransaction.vue]: Sure, Perhaps a better more descriptive name. `hide-right-side` doesn't tell me much if I'm not familiar with the component.

Perhaps `block-data`, `meta-data`…

Comment (thebkr7) [reply src/components/transactions/LiAnyTransaction.vue]: Sure, but what data is it blocking would come to mind with those names. Let's chat in the call tomorrow.

Comment (jbibla) [reply src/components/transactions/LiAnyTransaction.vue]: sorry guys, i think this was my suggestion 

yeah, let's chat about it tomorrow.

**Pull request merged**